tensorrt_llm/_torch/flashinfer_utils.py

```
@@ -7,12 +7,54 @@
+    if raw in ("0", "false", "off", "no"):
+        return False
+
+    if not _flashinfer_pdl_supported_on_current_device():
+        # Distinguish explicit TRTLLM_ENABLE_PDL=1 on Ampere (warn once) vs default path (info once).
+        env_explicit = "TRTLLM_ENABLE_PDL" in os.environ
+        if env_explicit and raw in ("1", "true", "on", "yes"):
+            if not getattr(get_env_enable_pdl, "_unsupported_warned", False):
+                logger.warning(
+                    "TRTLLM_ENABLE_PDL requests PDL but this GPU is below sm_90; "
+                    "FlashInfer kernels will run with PDL disabled.")
+                setattr(get_env_enable_pdl, "_unsupported_warned", True)
```

Ruff (0.15.10)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)